### PR TITLE
[Feat] L402 Explorer: Interact with L402 APIs via Web Interface

### PR DESCRIPTION
**Pull request by priyanshiiit**
## Description

This PR adds the L402 Explorer, a tool designed to simplify the testing of L402 APIs. Currently, testing these APIs from the command line or existing software can be cumbersome. With the L402 Explorer, developers can easily explore L402 APIs directly from their browsers. 

## Screenshots/GIF

| Screenshot 2024-04-14 193011 | Screenshot 2024-04-14 193032 |
| --- | --- |
| Screenshot 2024-04-14 193052 | Screenshot 2024-04-13 2307322 |

## Testing
Feature can be tested at: https://l402-explorer.vercel.app/

### CODE REVIEW

**Comment by priyanshiiit** [comment]
Hi @SamuelAl, Thanks for sharing the task!
The PR is ready to review.

File: Dockerfile
Comment (SamuelAl) [inline Dockerfile:1]: This works well for testing, but production would require a more production-ready Docker build. Check-out https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile

File: public/next.svg
Comment (SamuelAl) [inline public/next.svg:1]: Please delete if not needed.

File: public/vercel.svg
Comment (SamuelAl) [inline public/vercel.svg:1]: Please delete if not needed.

File: src/pages/api/hello.ts
Comment (SamuelAl) [inline src/pages/api/hello.ts:1]: We can remove this.

File: src/utils/index.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const invoiceMatch = header?.match(invoiceRegex);
+
+  return {
+    macaroon: macaroonMatch ? macaroonMatch[1] : null,
+    invoice: invoiceMatch ? invoiceMatch[1] : null,
+  };
+};
+
+export const isValidUrl = (url: string): boolean => {
+  const pattern = /^(ftp|http|https):\/\/[^ "]+$/;
```
Comment (SamuelAl) [inline src/utils/index.tsx:15]: would have been good to allow for URLs without https:// and add it automatically, but this has complications of its own so this approach is acceptable for an initial MVP.

**Review by SamuelAl — CHANGES_REQUESTED** [review]
Good job! Just some small comments to clean up, but overall really impressive.

**Comment by priyanshiiit** [comment]
Thanks for review. Updated!

**Review by SamuelAl — APPROVED** [review]
LGTM. Well done!

**Comment by SamuelAl** [comment]
@priyanshiiit  to claim the bounty, please send a lightning invoice for 0.00114 BTC (70 EUR) to samuel.alarco@sulusolutions.com . Please make sure it is valid for at least 1 day to allow for processing of the payment on our end.

**Pull request merged**